project.py

```diff
@@ -0,0 +1,153 @@
+        if "peter" in x.lower():
+            speech[index] = "Trump"
+        elif "lois" in x.lower():
+            speech[index] = "Ivanka"
+        elif "brian" in x.lower():
+            speech[index] = "Pence"
+        elif "stewie" in x.lower():
+            speech[index] = "Obama"
+        elif "quagmire" in x.lower():
+            speech[index] = "ISIS"
+        elif "meg" in x.lower():
+            speech[index] = "Mexico"
+        elif "chris" in x.lower():
+            speech[index] = "China"
+        elif "quahog" in x.lower():
+            speech[index] = "MURICA"
+        elif "griffin" in x.lower():
+            speech[index] = "President"
+        if '.' in x:
+            temp = x
+            mini = temp.split('.')
+            speech[index] = mini[0]
+            speech.insert(index, mini[1].capitalize())
+        index += 1
+    index = 0
+    for y in speech:
+        if " " in y:
+            speech[index] = y.replace(' ', '')
+        index += 1
+    if ('.' in speech[-1]) or ('!' in speech[-1]) or ('?' in speech[-1]):   #how to determine if punctuation needs to be added at the end
+        product = ' '.join(speech)
+    else:
+        product = ' '.join(speech) + '.'
+    return product #return a string of the final speech.
+
+if __name__ == '__main__':
+    allscripts = ''
+    my_file = Path("dict1.p")
+    my_file2 = Path("dict2.p")
+    my_file3 = Path("allwords.p")
+    # print(not my_file.is_file()) #For testing purposes
+    # print(not my_file2.is_file()) #For testing purposes
+    # print(not my_file3.is_file()) #For testing purposes
+    if (not my_file.is_file()) or (not my_file2.is_file()) or (not my_file3.is_file()): #check to see if there are already dictionaries; if there aren't then dictionaries are created.
+        basehtml = 'http://www.springfieldspringfield.co.uk/view_episode_scripts.php?tv-show=family-guy&episode=s'
+        temptml = basehtml
+        # for mini in ([8 '01e'], [22 '02e'], [23 '03e'], [28 '04e'], [19 '05e'], [13 '06e'], [17 '07e'], [22 '08e'], [19 '09e'], [24 '10e'], [23 '11e'], [22 '12e'], [19 '13e'], [21 '14e'], [14 '15e'])
+        #The above line is old code that I can reference to draw upon more family guy seasons
+        for i in range (1,8):   #season1 of family guy
+            temp =  temptml + '01e' + '0%d' % (i)
+            html = BeautifulSoup(requests.get(temp).text, 'lxml')
+            thing = html.find("div", class_ = 'scrolling-script-container')
+            allscripts = allscripts + " " + str(thing.text)
+        for i in range (1,22):   #season2
+            if i < 10:
+                temp =  temptml + '02e' + '0%d' % (i)
+            else:
+                temp =  temptml + '02e' + '%d' % (i)
+            html = BeautifulSoup(requests.get(temp).text, 'lxml')
+            thing = html.find("div", class_ = 'scrolling-script-container')
+            allscripts = allscripts + " " + str(thing.text)
+
+        for i in range (1,22):   #season3
```

Comment:
To cut down on repetitive code, you can make a tuple like `episodes_in_season = (8, 22, 22)` and then:
```python
for episodes in episodes_in_season:
    for i in range episodes:
        # the code you have here
```
That way you get rid of magic numbers AND you don't have to copy/paste code